### DISCUSSION TITLE
Very slow command run on iOS17?

### DISCUSSION THREAD

**Discussion by sunnyxx** [doronz88/pymobiledevice3#732] [Q&A]
Got very slow commands running on iOS 17, I don't know it's normal or not:

`python3 -m pymobiledevice3 remote start-tunnel -p quic --script-mode` takes about 1min
`python3 -m pymobiledevice3 developer dvt simulate-location set --rsd fdd8:d907:fd5c::1 59371 -- x y` takes about 1min
`python3 -m pymobiledevice3 developer dvt simulate-location clear --rsd fdd8:d907:fd5c::1 59371` takes about 30s

devices: iPhone 14 Pro Max iOS 17.2 / Mac Mini M2

**Comment by doronz88** [comment]
I cannot reproduce it. Please try to profile it yourself and provide more data what are the time-consuming operations

**Reply by bobbyc** [reply]
I do meet the same issue with @sunnyxx. When I set simulate-location, the command almost take 1 min to take effect on iPhone. 

device: iPhone Pro Max iOS 17.2 and iPhone 11 iOS 17.2 / Mac Mini M2.

@doronz88 Could you share how to debug this issue? Or any log I can help to collect.

**Reply by doronz88** [reply]
I don't have anything smart to tell except adding prints and viewing the device syslog

**Reply by bobbyc** [reply]
@doronz88 As your exprience to work iOS 17, what is the behavior or latency at your side? just few seconds? 

Just wondering, looks like secure tunnel is over IPv6, if it is possible caused by anti-malware software, firewall, or content filtering?

**Reply by doronz88** [reply]
not possible. usually it takes on my device up to 1s

**Reply by bobbyc** [reply]
https://developer.apple.com/forums/thread/737875?answerid=768846022#768846022

This looks like the issue and the solutions. But it doesn't work for me.

**Reply by sunnyxx** [reply]
Try this: https://github.com/iOSRealRun/iOSRealRun-cli-17
It works very fast when using pymobiledevice3 as lib with python (rather than cmd in shell like me using JavaScript)
And after I run this python script once, my original JS script worked magically !?, don't know why, I tried to reboot my Mac and my iPhone, still worked :)
@bobbyc you should try it
